tests/queries/0_stateless/01846_null_as_default_for_insert_select.sql

```diff
@@ -0,0 +1,15 @@
+DROP TABLE IF EXISTS test_null_as_default;
+CREATE TABLE test_null_as_default (s String DEFAULT 'WORLD') ENGINE = Memory;
+
+INSERT INTO test_null_as_default SELECT 'HELLO' UNION ALL SELECT NULL;
+SELECT * FROM test_null_as_default ORDER BY s;
+SELECT '';
+
+INSERT INTO test_null_as_default SELECT NULL;
+SELECT * FROM test_null_as_default ORDER BY s;
+SELECT '';
+
+REPLACE TABLE test_null_as_default (s String DEFAULT 'WORLD', ss String DEFAULT 'PEOPLE') ENGINE = Memory;
```

Comment:
Also it is good to add cases:
* with different types, like `x UInt64, y UInt8 default x - 10 `
* with recursive defaults, like `x UInt32, y UInt32 default z + 1, z UInt32  default x + 1` (and insert into (x, y))